src/components/Navbar/components/Wallet/WalletButton.tsx

```diff
@@ -2,8 +2,10 @@ import { WarningOutlined } from '@ant-design/icons'
 import { Trans } from '@lingui/macro'
 import { Button } from 'antd'
 import { useWallet } from 'hooks/Wallet'
+import LegalNoticeModal from 'components/modals/LegalNoticeModal'
```

Comment:
I wonder if this behaviour should be a wallet signing request instead. NAL, but I am not sure how usable this is as a legal notice. That said, im unsure how usable the wallet signing is either, but we would have a record to point to at very least 🤷‍♂️